W-21080560 Adding launch.json properties reference topic (#677)
mulesoft/docs-code-builder: code-builder-home/modules/ROOT/pages/ref-launch-json-properties.adoc

| `mule.runtime.args`
| No
| Command-line arguments passed to the Mule runtime JVM. Use the `-M-D` prefix for system properties. +
 +
Examples: +
`-M-Dencryption.key=myKey` (encryption key) +
`-M-Denv=dev` (environment selector) +
`-M-Dhttp.port=8082` (HTTP listener port) +
 +
For more information, see xref:int-run-mule-apps-with-properties.adoc[] and xref:int-create-secure-configs.adoc[].

| `mule.debug.port`
| No
| Port number for the Mule debugger. Default: `6666`. Change this value when running multiple applications concurrently to avoid port conflicts. Example: `6667`, `6668`.

| `noDebug`
| No
| Set to `true` to run the application without attaching the debugger. Useful for performance testing or when you don't need breakpoint debugging. Default: `false`.

|===

Either `mule.project` or `mule.projects` must be specified, but not both.

[[multiple-configs]]
== Creating Multiple Configuration Profiles

Create multiple launch configurations to test different environments without modifying your application code. Each configuration can have different runtime arguments, debug ports, and environment settings.

> The following example shows three configurations for Local Development, QA, and a Mocked environment:

Reviewer: ```suggestion
This example shows three configurations for Local Development, QA, and a Mocked environment:
```

      "mule.project": "${workspaceFolder}",
      "mule.home": "${config:mule.homeDirectory}",
      "mule.runtime.args": "${config:mule.runtime.defaultArguments} -M-Denv=dev -M-Dencryption.key=devKey123",
      "mule.debug.port": 6666
    },
    {
      "type": "mule-xml-debugger",
      "request": "launch",
      "name": "Debug - QA Environment",
      "mule.project": "${workspaceFolder}",
      "mule.home": "${config:mule.homeDirectory}",
      "mule.runtime.args": "${config:mule.runtime.defaultArguments} -M-Denv=qa -M-Dencryption.key=qaKey456",
      "mule.debug.port": 6667
    },
    {
      "type": "mule-xml-debugger",
      "request": "launch",
      "name": "Run - Mocked Services",
      "mule.project": "${workspaceFolder}",
      "mule.home": "${config:mule.homeDirectory}",
      "mule.runtime.args": "${config:mule.runtime.defaultArguments} -M-Denv=mock -M-Dmock.services=true",
      "mule.debug.port": 6666,
      "noDebug": true
    }
  ]
}
----

=== Switch Between Configurations

To switch between run configurations:

. Open the *Run and Debug* view from the activity bar (image:icon-run-debug.png["",18,18]).
. Click the configuration dropdown in the debug panel.

> . Select the configuration to use

Reviewer: ```suggestion
. Select the configuration to use.
```
